be/src/pipeline/pipeline_x/pipeline_x_fragment_context.cpp

```diff
@@ -453,41 +456,62 @@ Status PipelineXFragmentContext::_build_pipeline_tasks(
+            runtime_state->set_query_mem_tracker(_query_ctx->query_mem_tracker);
 
-        _runtime_states[i] = RuntimeState::create_unique(
-                local_params.fragment_instance_id, request.query_id, request.fragment_id,
-                request.query_options, _query_ctx->query_globals, _exec_env);
-        if (local_params.__isset.runtime_filter_params) {
-            _runtime_states[i]->set_runtime_filter_params(local_params.runtime_filter_params);
-        }
-        _runtime_states[i]->set_query_ctx(_query_ctx.get());
-        _runtime_states[i]->set_query_mem_tracker(_query_ctx->query_mem_tracker);
+            runtime_state->set_query_ctx(_query_ctx.get());
+            runtime_state->set_be_number(local_params.backend_num);
 
-        static_cast<void>(_runtime_states[i]->runtime_filter_mgr()->init());
-        _runtime_states[i]->set_be_number(local_params.backend_num);
+            if (request.__isset.backend_id) {
+                runtime_state->set_backend_id(request.backend_id);
+            }
+            if (request.__isset.import_label) {
+                runtime_state->set_import_label(request.import_label);
+            }
+            if (request.__isset.db_name) {
+                runtime_state->set_db_name(request.db_name);
+            }
+            if (request.__isset.load_job_id) {
+                runtime_state->set_load_job_id(request.load_job_id);
+            }
 
-        if (request.__isset.backend_id) {
-            _runtime_states[i]->set_backend_id(request.backend_id);
-        }
-        if (request.__isset.import_label) {
-            _runtime_states[i]->set_import_label(request.import_label);
-        }
-        if (request.__isset.db_name) {
-            _runtime_states[i]->set_db_name(request.db_name);
-        }
-        if (request.__isset.load_job_id) {
-            _runtime_states[i]->set_load_job_id(request.load_job_id);
-        }
+            runtime_state->set_desc_tbl(_desc_tbl);
+            runtime_state->set_per_fragment_instance_idx(local_params.sender_id);
+            runtime_state->set_num_per_fragment_instances(request.num_senders);
+            runtime_state->resize_op_id_to_local_state(max_operator_id(), max_sink_operator_id());
+            runtime_state->set_load_stream_per_node(request.load_stream_per_node);
+            runtime_state->set_total_load_streams(request.total_load_streams);
+            runtime_state->set_num_local_sink(request.num_local_sink);
+            DCHECK(runtime_filter_mgr);
+            runtime_state->set_pipeline_x_runtime_filter_mgr(runtime_filter_mgr.get());
+        };
+        // build instance runtime state
+        _runtime_states[i] = RuntimeState::create_unique(
+                this, local_params.fragment_instance_id, request.query_id, request.fragment_id,
+                request.query_options, _query_ctx->query_globals, _exec_env);
+        // build runtime_filter_mgr for each instance
+        runtime_filter_mgr =
```

Comment:
Do not pass `_runtime_states[i].get()` in this constructor because this runtime state is instance's level